test/suite/test_prepare01.py

```diff
@@ -0,0 +1,177 @@
+    # checkpoint matches the expected value.
+    def check_checkpoint(self, expected):
+        s = self.conn.open_session()
+        s.checkpoint("name=test")
+        cursor = s.open_cursor(self.uri, None, "checkpoint=test")
+        self.assertEqual(self.cursor_count(cursor), expected)
+        s.close()
+
+    # Open a cursor with snapshot isolation, and assert the number of records
+    # visible to the cursor matches the expected value.
+    def check_txn_cursor(self, level, expected):
+        s = self.conn.open_session()
+        cursor = s.open_cursor(self.uri, None)
+        s.begin_transaction(level)
+        self.assertEqual(self.cursor_count(cursor), expected)
+        s.close()
+
+    # Open a session with snapshot isolation, and assert the number of records
+    # visible to the cursor matches the expected value.
+    def check_txn_session(self, level, expected):
+        s = self.conn.open_session(level)
+        cursor = s.open_cursor(self.uri, None)
+        s.begin_transaction()
+        self.assertEqual(self.cursor_count(cursor), expected)
+        s.close()
+
+    def check(self, cursor, committed, total):
+        # The cursor itself should see all of the records.
+        if cursor != None:
+            cursor.reset()
+            self.assertEqual(self.cursor_count(cursor), total)
+
+        # Read-uncommitted should see all of the records.
+        # Snapshot and read-committed should see only committed records.
+        self.check_txn_cursor('isolation=read-uncommitted', total)
+        self.check_txn_session('isolation=read-uncommitted', total)
+
+        self.check_txn_cursor('isolation=snapshot', committed)
+        self.check_txn_session('isolation=snapshot', committed)
+
+        self.check_txn_cursor('isolation=read-committed', committed)
+        self.check_txn_session('isolation=read-committed', committed)
+
+        # Checkpoints should only write committed items.
+        self.check_checkpoint(committed)
+
+    # Loop through a set of inserts, periodically committing; before each
+    # commit, verify the number of visible records matches the expected value.
+    def test_visibility(self):
+        self.session.create(self.uri,
+            'key_format=' + self.key_format +
+            ',value_format=' + self.value_format)
+
+        committed = 0
+        cursor = self.session.open_cursor(self.uri, None)
+        self.check(cursor, 0, 0)
+        msg = "/prepare_transaction is not supported/"
+        self.session.begin_transaction()
+        for i in xrange(self.nentries):
+            if i > 0 and i % (self.nentries / 37) == 0:
+                self.check(cursor, committed, i)
+                self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
+                    lambda: self.session.prepare_transaction(), msg)
+                self.session.commit_transaction()
+                committed = i
+                self.session.begin_transaction()
+
+            if self.key_format == 'S':
```

Comment:
Frankly I am not sure of the logic either. I copied test_txn01.py, so that the prepare_transaction can be tested that it is not breaking any basic transaction properties.